web/auth.go

```diff
@@ -413,79 +423,124 @@ type tokenResponse struct {
+func getBearerToken(req *http.Request) string {
+	authHeader := req.Header.Get("Authorization")
+	if authHeader == "" {
+		return ""
+	}
 
-	_, span := trace.StartSpan(ctx, r.URL.Path)
-	defer span.End()
+	parts := strings.Split(authHeader, " ")
+	if len(parts) == 2 && parts[0] == "Bearer" {
+		return parts[1]
+	}
 
-	token, err := r.Cookie("viam.auth.token")
-	if HandleError(w, err, h.logger, "getting token cookie") {
-		return
+	return ""
+}
+
+// getAuthCookieValues reads the authentication cookie values as a /token response.
+func getAuthCookieValues(r *http.Request) *tokenResponse {
+	token, err := r.Cookie(ViamTokenCookie)
+	if err != nil || token.Value == "" {
+		return nil
 	}
 
-	refresh, err := r.Cookie("viam.auth.refresh")
-	if HandleError(w, err, h.logger, "getting refresh cookie") {
-		return
+	refresh, err := r.Cookie(ViamRefreshCookie)
+	// TODO: Check if refresh is empty when implemented, always empty now
+	if err != nil {
+		return nil
 	}
 
-	expiry, err := r.Cookie("viam.auth.expiry")
-	if HandleError(w, err, h.logger, "getting expiry cookie") {
-		return
+	expiry, err := r.Cookie(ViamExpiryCookie)
+	if err != nil || expiry.Value == "" {
+		return nil
 	}
 
-	response := &tokenResponse{
+	return &tokenResponse{
 		AccessToken:  token.Value,
 		RefreshToken: refresh.Value,
 		Expiry:       expiry.Value,
 	}
+}
 
-	w.Header().Set("Content-Type", "application/json")
-	data, err := json.Marshal(response)
-	if err != nil {
-		temp := errors.New("failed to verify marshal token data: " + err.Error())
-		w.WriteHeader(http.StatusInternalServerError)
-		_, err = w.Write([]byte(temp.Error()))
-		if err != nil {
-			utils.UncheckedError(err)
-		}
-		h.logger.Error(temp)
-		return
-	}
-
+// clearAuthCookies removes auth cookies from /callback
+// - to be used after reading the cookies so they can only be used once.
+func clearAuthCookies(w http.ResponseWriter) {
 	http.SetCookie(w, &http.Cookie{
-		Name:     "viam.auth.token",
+		Name:     ViamTokenCookie,
 		Value:    "",
 		Path:     "/",
 		MaxAge:   -1,
-		Secure:   r.TLS != nil,
 		SameSite: http.SameSiteLaxMode,
 		HttpOnly: true,
 	})
 
 	http.SetCookie(w, &http.Cookie{
-		Name:     "viam.auth.refresh",
+		Name:     ViamRefreshCookie,
 		Value:    "",
 		Path:     "/",
 		MaxAge:   -1,
-		Secure:   r.TLS != nil,
 		SameSite: http.SameSiteLaxMode,
 		HttpOnly: true,
 	})
 
 	http.SetCookie(w, &http.Cookie{
-		Name:     "viam.auth.expiry",
+		Name:     ViamExpiryCookie,
 		Value:    "",
 		Path:     "/",
 		MaxAge:   -1,
-		Secure:   r.TLS != nil,
 		SameSite: http.SameSiteLaxMode,
 		HttpOnly: true,
 	})
+}
 
-	_, err = w.Write(data)
-	utils.UncheckedError(err)
+func (h *tokenHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	ctx, cancel := context.WithTimeout(r.Context(), 10*time.Second)
+	defer cancel()
+
+	_, span := trace.StartSpan(ctx, r.URL.Path)
+	defer span.End()
+
+	current := getBearerToken(r)
+	data := getAuthCookieValues(r)
+	clearAuthCookies(w)
```

Comment:
[Minor] We might want to combine these into a single `getAndClearAuthCookies` (or whatever name to communicate that the cookies are one-time-use) to make this difficult to accidentally break in the future

Reply:
Makes sense